kadas/gui/search/kadasalternategotolocatorfilter.cpp

```diff
@@ -131,39 +131,34 @@ void KadasAlternateGotoLocatorFilter::fetchResults( const QString &string, const
 
   if ( !match.hasMatch() )
   {
-    // Check if the string is a pair of degree minute second
-    thread_local QString dmsRx = QStringLiteral( R"(\d{1,3}(?:[^0-9.]+[0-5]?\d)?[^0-9.]+[0-5]?\d(?:[\.\%1]\d+)?)" ).arg( locale.decimalPoint() );
-    thread_local QRegularExpression separatorRx4( QStringLiteral(
-                                                    "^("
-                                                    R"((\s*%1[^0-9.,]*(?:\s*%2)?[-+NSEWnsew]?)[,\s]+(%1[^0-9.,]*(?:\s*%2)?[-+NSEWnsew]?))"
-                                                    ")|("
-                                                    R"(((?:([-+NSEWnsew])\s*)%1[^0-9.,]*(?:\s*%2)?)[,\s]+((?:([-+NSEWnsew])\s*)%1[^0-9.,]*(?:\s*%2)?))"
-                                                    ")$"
-    )
-                                                    .arg( dmsRx, degreeSymbol ) );
+    // Improved regex for DMS: matches e.g. 7°34'25.5"E,46°54'98.3"N or 7 34 25.5 E, 46 54 98.3 N
+    thread_local QRegularExpression separatorRx4(
+      R"(^\s*([0-9]{1,3})[°º\s]\s*([0-9]{1,2}(?:[\.,][0-9]+)?)\s*['’′]?\s*([0-9]{1,2}(?:[\.,][0-9]+)?)?\s*["”″]?\s*([NSEWnsew])[\s,;]+([0-9]{1,3})[°\s]\s*([0-9]{1,2}(?:[\.,][0-9]+)?)\s*['’′]?\s*([0-9]{1,2}(?:[\.,][0-9]+)?)?\s*["”″]?\s*([NSEWnsew])\s*(:?\d+)?$)"
```

Comment:
```suggestion
      R"(^\s*"         // leading space (optional)
      R"([0-9]{1,3})[°º\s]\s*" // first (lat) degrees
      R"([0-9]{1,2}(?:[\.,][0-9]+)?)\s*['’′]?\s*" // first (lat) minutes
      R"([0-9]{1,2}(?:[\.,][0-9]+)?)?\s*["”″]?\s*" // first (lat) seconds
      R"([NSEWnsew])" // first direction
      R"[\s,;]+" // separator
      R"([0-9]{1,3})[°\s]\s*" // second (lon) degrees
      R"([0-9]{1,2}(?:[\.,][0-9]+)?)\s*['’′]?\s*" // second (lon) minutes
      R"([0-9]{1,2}(?:[\.,][0-9]+)?)?\s*["”″]?\s*" // second (lon) seconds
      R"([NSEWnsew])\s*" // second direction
      R"(:?\d+)?$)" // altitude (optional)
```

readability suggestion (I hope I got that right)